modules/punctuation/punctuation.cpp

```diff
@@ -395,4 +398,54 @@ Punctuation::getPunctuation(const std::string &language, uint32_t unicode) {
     return iter->second.getPunctuation(unicode);
 }
 
+const fcitx::Configuration *
+Punctuation::getSubConfig(const std::string &path) const {
+    if (path == "punctuationmap-zh_CN") {
+        return &punctuationMapConfig_;
+    }
+    return nullptr;
+}
+
+void Punctuation::setupPunctuationMapConfig(bool isSyncToDefault) {
+    auto configValue = punctuationMapConfig_.entries.mutableValue();
+    auto puncMap = profiles_["zh_CN"].getPunctuationMap();
+    for (auto &[key, value] : *puncMap) {
+        PunctuationMapEntryConfig entryConfig;
+        std::string punc(1, (char)key);
+        entryConfig.original.setValue(punc);
+        entryConfig.mapResult.setValue(value.first);
+        configValue->emplace_back(entryConfig);
+    }
+    if (isSyncToDefault) {
```

Comment:
You should think of remove this argument, and directly call syncDefaultValueToCurrent in the current (true) caller.